support-frontend/app/services/stepfunctions/Client.scala

```diff
@@ -31,22 +31,27 @@ object Client {
 
 class Client(client: AWSStepFunctionsAsync, arn: StateMachineArn) extends SafeLogging {
 
-  private def startExecution(arn: String, input: String)(implicit
+  private def startExecution(arn: String, input: String, name: String)(implicit
       ec: ExecutionContext,
   ): Response[StartExecutionResult] = convertErrors {
-    AwsAsync(client.startExecutionAsync, new StartExecutionRequest().withStateMachineArn(arn).withInput(input))
+    val startExecutionRequest =
+      new StartExecutionRequest()
+        .withStateMachineArn(arn)
+        .withInput(input)
+        .withName(name + "-" + System.nanoTime().toString)
```

Comment:
has to be unique for the last 90 days hence the nanotime

Reply:
Could we use [`UUID.randomUUID()`](https://www.baeldung.com/java-uuid)?

Reply:
we could, I didn't do that because it's unnecessariuly long